dev-tools/xtask/src/a4x2_package.rs

```diff
@@ -0,0 +1,382 @@
+        );
+        assert!(
+            treeish.chars().all(|c| c.is_ascii_hexdigit()),
+            "treeish should be a 40-character commit hash"
+        );
+        treeish
+    };
+
+    // Clone the local source tree into the packaging work dir
+    let _popdir = sh.push_dir(&env.omicron_dir);
+    cmd!(sh, "git init").run()?;
+
+    let src_dir = &env.src_dir;
+    cmd!(sh, "git remote add origin {src_dir}").run()?;
+
+    cmd!(sh, "git fetch origin {treeish}").run()?;
+    cmd!(sh, "git checkout {treeish}").run()?;
+
+    // XXX is this needed in CI?
+    cmd!(sh, "./tools/install_builder_prerequisites.sh -yp").run()?;
+
+    Ok(())
+}
+
+fn build_end_to_end_tests(sh: &Shell, env: &Environment) -> Result<()> {
+    // I'm not confident this does what it should, because I don't know much
+    // about the end to end tests. This is just going from what the shell script
+    // did.
+    cmd!(sh, "banner 'end to end'").run()?;
+    let _popdir = sh.push_dir(&env.omicron_dir);
+
+    let cargo = &env.cargo;
+    cmd!(sh, "{cargo} build -p end-to-end-tests --bin commtest --bin dhcp-server --release").run()?;
+
+    let end_to_end_dir = env.out_dir.join("end-to-end-tests");
+    sh.create_dir(&end_to_end_dir)?;
+    sh.copy_file("target/release/commtest", &end_to_end_dir)?;
+    sh.copy_file("target/release/dhcp-server", &end_to_end_dir)?;
+
+    Ok(())
+}
+
+/// Generate a bundle with the live tests included and ready to go.
+fn build_live_tests(sh: &Shell, env: &Environment) -> Result<()> {
+    cmd!(sh, "banner 'live tests'").run()?;
+    let _popdir = sh.push_dir(&env.omicron_dir);
+
+    // We need nextest available, both to generate the test bundle and to
+    // include in the output tarball to execute the bundle on a4x2.
+
+    // XXX right now we only download a new nextest in CI, and we use
+    // the user's locally installed nextest otherwise. I'm inclined to
+    // say we should always download a fresh one for consistency.
+    // thoughts?
+    let nextest_path = if env.in_ci {
+        // In CI, we need to download nextest
+        let config_str = sh.read_file(".config/nextest.toml")?;
+        let nextest_config: NextestConfig = toml::from_str(&config_str)?;
+
+        // This line of bash comes from build-and-test.sh
+        // If you want to rewrite this in rust, please! Make sure you
+        // - retry correctly
+        // - follow redirects correctly
+        // - handle http errors
+        cmd!(sh, "bash -c")
+            .arg(&format!("curl -sSfL --retry 10 https://get.nexte.st/'{}'/illumos | gunzip | tar -xvf - -C ~/.cargo/bin", nextest_config.nextest_version.recommended))
+            .run()?;
+        env.home_dir.join(".cargo/bin/cargo-nextest")
+    } else {
+        // XXX do we need to test nextest exists & is the right version? need to
+        // see if the incidental errors from stuff below already take care of
+        // telling someone outside of CI what they need to be doing.
```

Comment:
You could run `cargo nextest show-config version` -- [exit code 92](https://docs.rs/nextest-metadata/latest/nextest_metadata/enum.NextestExitCode.html#associatedconstant.REQUIRED_VERSION_NOT_MET) means that the required version isn't met, and [exit code 10](https://docs.rs/nextest-metadata/latest/nextest_metadata/enum.NextestExitCode.html#associatedconstant.RECOMMENDED_VERSION_NOT_MET) means that the recommended version isn't met.